wms/models/datasets/ugrid.py

```diff
@@ -284,7 +281,29 @@ def getmap(self, layer, request):
                     return self.empty_response(layer, request)
 
                 if request.GET['image_type'] == 'filledcontours':
-                    return mpl_handler.tricontourf_response(tri_subset, data, request)
+                    if np.isnan(np.sum(data)):
+                        data_mask = np.logical_not(np.isnan(data))  # mask for non-NaN data elements
```

Comment:
@ocefpaf,

To answer your first question, I've seen little performance difference between `~np.isnan(data)` and `np.logical_not(np.isnan(data)`. My poking suggests they perform identically.

Anyway, I like the the suggestion of calling `np.blah` fewer times.